### ISSUE TITLE
ApiTag: Public object `...` is missing @api tag error on Data class attribute

### ISSUE THREAD

**Issue by wilsonsilva** [mensfeld/yard-lint#128]
I'm getting an error about a missing @api error on a Data class attribute. I tried to use `!@method` and it didn't work either.

## Errors
```sh
Inspecting with 27 validators
Found 3 offense(s):

[E] /Users/wilson/projects/rb/jsonrpc-middleware/lib/jsonrpc/configuration.rb:53
    ApiTag: Public object `JSONRPC::Configuration::Procedure#allow_positional_arguments` is missing @api tag

[E] /Users/wilson/projects/rb/jsonrpc-middleware/lib/jsonrpc/configuration.rb:53
    ApiTag: Public object `JSONRPC::Configuration::Procedure#contract` is missing @api tag

[E] /Users/wilson/projects/rb/jsonrpc-middleware/lib/jsonrpc/configuration.rb:53
    ApiTag: Public object `JSONRPC::Configuration::Procedure#parameter_name` is missing @api tag
```

## Source
https://github.com/wilsonsilva/jsonrpc-middleware/blob/77f95a5d13a92b0f5a15038def171752f910f7ce/lib/jsonrpc/configuration.rb#L19-L53

```ruby
# frozen_string_literal: true

module JSONRPC
  # Configuration class for JSON-RPC procedure management and validation.
  #
  # @api public
  #
  # This class provides functionality to register, retrieve, and validate JSON-RPC procedures.
  # It acts as a central registry for method definitions and their parameter constraints.
  #
  # @example Registering a procedure
  #   JSONRPC::Configuration.instance.procedure('sum') do
  #     params do
  #       required(:numbers).value(:array, min_size?: 1)
  #     end
  #   end
  #
  class Configuration
    # Represents a registered JSON-RPC procedure with its validation contract and configuration
    #
    # @api public
    #
    # @!attribute [r] allow_positional_arguments
    #   Indicates if the procedure accepts positional arguments
    #
    #   @api public
    #
    #   @example
    #     procedure.allow_positional_arguments # => true
    #
    #   @return [Boolean] whether the procedure accepts positional arguments
    #
    # @!attribute [r] contract
    #   The validation contract for procedure parameters
    #
    #   @api public
    #
    #   @example
    #     procedure.contract # => #<Dry::Validation::Contract...>
    #
    #   @return [Dry::Validation::Contract] the validation contract for procedure parameters
    #
    # @!attribute [r] parameter_name
    #   The name of the first parameter in the contract schema
    #
    #   @api public
    #
    #   @example
    #     procedure.parameter_name # => :numbers
    #
    #   @return [Symbol, nil] the name of the first parameter in the contract schema
    #
    Procedure = Data.define(:allow_positional_arguments, :contract, :parameter_name)

    # ...
  end
end
```

## Yard result
```sh
Files:          24
Modules:         5 (    0 undocumented)
Classes:        20 (    0 undocumented)
Constants:       2 (    0 undocumented)
Attributes:     18 (    0 undocumented)
Methods:        78 (    0 undocumented)
100.00% documented
```

## Environment
`yard` version `0.9.39`
`yard-lint` version `1.5.1`
`ruby` version `4.0.2`

**Comment by mensfeld** [comment]
Let me see what I can do with that one 🙏

**Comment by mensfeld** [comment]
Great catch. Thank you. I will try to release it next week.

**Comment by wilsonsilva** [comment]
Thank you, @mensfeld! Great timing :)

**Comment by mensfeld** [comment]
Anytime. It's just hard to tackle all the edge cases of all the ruby code in the world 😅 but we will get there!
